tx/textual/valuerenderer/duration.go

```diff
@@ -0,0 +1,217 @@
+package valuerenderer
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"google.golang.org/protobuf/reflect/protoreflect"
+	dpb "google.golang.org/protobuf/types/known/durationpb"
+)
+
+type durationValueRenderer struct{}
+
+// NewDurationValueRenderer returns a ValueRenderer for protocol buffer Duration messages.
+// It renders durations by grouping seconds into units of days (86400s), hours (3600s),
+// and minutes(60s), plus the total seconds elapsed. E.g. a duration of 1483530s is
+// formatted as "17 days, 4 hours, 5 minutes, 30 seconds (1483530 seconds total)".
+// Note that the days are always 24 hours regardless of daylight savings changes.
+func NewDurationValueRenderer() ValueRenderer {
+	return durationValueRenderer{}
+}
+
+const (
+	min_sec  = 60
+	hour_sec = 60 * min_sec
+	day_sec  = 24 * hour_sec
+)
+
+type factors struct {
+	days, hours, minutes, seconds int
+}
+
+func factorSeconds(x int64) factors {
+	var f factors
+	f.days = int(x / int64(day_sec))
```

Comment:
I see Github's code scanning error "Potential integer overflow by integer type conversion", and I believe they are right (on archs where int == int32).

A safe solution would be to change `factors.{days,hours,minutes,seconds}` all to int64

Reply:
The code is safe because of restrictions on integer ranges that aren't expressed in the type system. But changing to `int64`, even though unnecessary, declutters the code and avoids a fight with code analysis. Fixed, here and below.